### PR TITLE
Ideal air loads

### PR DESCRIPTION
**Pull request by ChristopherCaradonna**
Pull request overview
---------------------

This PR adds an EUSS measure for ideal air loads.

### Pull Request Author: Chris CaraDonna

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [x] Upgrade Measures
 - [ ] Reporting Measures
 - [ ] Postprocessing

Author pull request checklist:

 - [ ] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Added tests for new measures
 - [ ] Updated measure .xml(s)
 - [ ] Register values added to `comstock_column_definitions.csv`
 - [ ] Both `options_lookup.tsv` files updated
 - [ ] 10k+ test run
 - [ ] Change documentation written
 - [ ] Measure documentation written
 - [ ] ComStock documentation updated
 - [ ] Changes reflected in example `.yml` files
 - [ ] Changes reflected in `README.md` files
 - [ ] Added 'See ComStock License' language to first two lines of each code file
 - [ ] Implements corresponding measure tests and indexing path in `test/measure_tests.txt` or/and `test/resource_measure_tests.txt`
 - [ ] All new and existing tests pass the CI

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```

### CODE REVIEW

File: resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                                                        include_outdoor_air: true,
+                                                        enable_dcv: false,
+                                                        econo_ctrl_mthd: 'NoEconomizer',
+                                                        heat_recovery_type: 'None',
+                                                        heat_recovery_sensible_eff: 0.7,
+                                                        heat_recovery_latent_eff: 0.65,
+                                                        add_output_meters: false)
+
+    # validity checks
+    unless ideal_loads_objects
```
Comment (eringold) [inline resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb]: I don't think `model_add_ideal_air_loads` will ever return false - looks like it will be an empty array (thus true), or standards will throw an error. Maybe revise this check to `if ideal_air_loads.empty?`

File: resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.xml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  <schema_version>3.1</schema_version>
+  <name>upgrade_hvac_enable_ideal_air_loads</name>
+  <uid>f70d4a3d-8b93-4fef-b679-858fe1e05acc</uid>
+  <version_id>a0e3eacb-5341-4b71-9a05-fa253a922db1</version_id>
+  <version_modified>2024-07-23T14:05:58Z</version_modified>
+  <xml_checksum>4435983D</xml_checksum>
+  <class_name>UpgradeHvacEnableIdealAirLoads</class_name>
+  <display_name>UpgradeHvacEnableIdealAirLoads</display_name>
+  <description>This will enable Ideal Air Loads for all thermal zones in the model. This will result in disconnecting any pre-existing HVAC systems from the zones.</description>
+  <modeler_description>This doesn't clean up any air or plant loops that are left. It was developed to use on models without HVAC. If you use it in another way you may have to clean up remaining objects prior to running a simulation.</modeler_description>
```
Comment (eringold) [inline resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.xml:12]: Don't think this has been updated, since there's no `modelerDescription` in the measure.rb, and besides the measure calls `remove_hvac`. It would be nice to have an update measure description.

File: measures/comstock_sensitivity_reports/README.md
Comment (eringold) [inline measures/comstock_sensitivity_reports/README.md:1]: these changes (and those to upgrade_heat_pump_rtu measure) shouldn't be part of this PR

File: resources/tests/models/370_warehouse_psz_gas_6A/measures/upgrade_hvac_add_heat_pump_rtu 1/LICENSE.md
Comment (eringold) [inline resources/tests/models/370_warehouse_psz_gas_6A/measures/upgrade_hvac_add_heat_pump_rtu 1/LICENSE.md:1]: don't include test model run directory files in PR.

**Review by eringold — CHANGES_REQUESTED** [review]

File: measures/comstock_sensitivity_reports/README.md [hunk 3]
```diff
@@ -2,21 +2,16 @@
 
 ###### (Automatically generated documentation)
 
-# ComStock_Sensitivity_Reports
+#
```
Comment (mdahlhausen) [inline measures/comstock_sensitivity_reports/README.md:5]: Looks like there was an error in the .xml update; rerun

File: resources/tests/models/370_warehouse_psz_gas_6A/measures/upgrade_hvac_add_heat_pump_rtu 1/LICENSE.md
Comment (asparke2) [reply resources/tests/models/370_warehouse_psz_gas_6A/measures/upgrade_hvac_add_heat_pump_rtu 1/LICENSE.md:1]: Agree, remove all these test outputs.

File: resources/measures/upgrade_hvac_add_heat_pump_rtu/README.md
Comment (asparke2) [inline resources/measures/upgrade_hvac_add_heat_pump_rtu/README.md:1]: Why is the hp rtu measure being changed in this PR?

File: measures/comstock_sensitivity_reports/README.md
Comment (asparke2) [reply measures/comstock_sensitivity_reports/README.md:1]: Agree, remove those files from this PR

File: resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb [hunk 4]
```diff
@@ -0,0 +1,91 @@
+# dependencies
+require 'openstudio-standards'
+
+# start the measure
+class UpgradeHvacEnableIdealAirLoads < OpenStudio::Measure::ModelMeasure
+  # define the name that a user will see, this method may be deprecated as
+  # the display name in PAT comes from the name field in measure.xml
+  def name
+    return 'UpgradeHvacEnableIdealAirLoads'
```
Comment (asparke2) [inline resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb]: Should be "Upgrade HVAC Enable Ideal Air Loads"

File: resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  end
+
```
Comment (asparke2) [inline resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb:11]: Missing description() and modeler_description() methods

File: resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb (hunk 1 above)
Comment (asparke2) [reply resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb]: correct, an array is always "true", regardless of whether it is empty or not

File: resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                                                        add_output_meters: false)
+
+    # validity checks
+    unless ideal_loads_objects
+      runner.registerError('Failure in creating ideal loads objects.  See logs from [openstudio.model.Model]. Likely cause is an invalid schedule input or schedule removed from by another measure.')
+      return false
+    end
+
+    # runner register final conditions of model
+    runner.registerFinalCondition("The model has #{ideal_loads_objects.size} ideal air loads objects.")
```
Comment (asparke2) [inline resources/measures/upgrade_hvac_enable_ideal_air_loads/measure.rb:100]: add an NA check and associated registerAsNotApplicable(). Maybe if the size of ideal_loads_objects does not match the size of conditioned_zones, I think that would indicate an error, but if the number of conditioned_zones was zero, then it's NA. IDK how that would happen in ComStock unless the code to check for conditioned zones wasn't working right.

**Review by asparke2 — CHANGES_REQUESTED** [review]
@ChristopherCaradonna some stuff to remove here not related to ideal loads, and a few changes to the actual ideal loads measure.
